### PR TITLE
added docs for adding roles to your local backend

### LINKED ISSUE #8: write docs for local backend roles
write docs for adding roles to self, which allows the user to use features that require super_user or admin role

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
closes #8

### CODE REVIEW

File: docs/local-backend-roles/localroles.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Setting roles in your local backend
+
+Please make sure that you have setup the backend locally. Watch these videos if you have not done that: 
+- <a href="https://www.youtube.com/watch?v=haqPaPRrhPU">Video 1</a>
+- <a href="https://www.youtube.com/watch?v=2Ja5hH2nH1o&feature=youtu.be">Video 2</a>
```
Comment (shubham-y) [inline docs/local-backend-roles/localroles.md]: NIT: We can replace "Video 1" and "Video 2" by their respective YouTube video titles

File: docs/local-backend-roles/localroles.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Steps: 
+
+1. Make sure that your backend server is running. To run the server run `yarn dev` in your project directory. This will run your server on localhost:3000
+
+2. The RDS backend hits `/users/self` endpoint to get the data of the user.
+
+3. So let's try to acces our details by going to http://localhost:3000/users/self. This will give a 401 error, to solve this follow the step below.
+
+4. Copy this URL `https://github.com/login/oauth/authorize?client_id=<your-client-id>` and paste it somewhere, now go to your  `local.js` file and copy the clientId that is present there, and paste it in place of `<your-client-id>` in the URL provided above.
```
Comment (shubham-y) [inline docs/local-backend-roles/localroles.md]: Suggestion: We can also add a link to GitHub docs explaining how to get GitHub client ID directly from [your application's configuration page](https://github.com/settings/developers).

**Review by shubham-y — COMMENTED** [review]
@vinayak-trivedi Thank you for the documenting the process.
Please take a look at the suggestions

File: docs/local-backend-roles/localroles.md (hunk 1 above)
Comment (vinayak-trivedi) [reply docs/local-backend-roles/localroles.md]: will do that

File: docs/local-backend-roles/localroles.md (hunk 2 above)
Comment (vinayak-trivedi) [reply docs/local-backend-roles/localroles.md]: Do we really need to do that? Someone who has set up the backend has that in their local.js and people who have not set up the backend need to set it up first

Comment (shubham-y) [reply docs/local-backend-roles/localroles.md]: For developers who are just going though the docs before setting up the backend might not have the context of `local.js` file. I think for them GitHub docs might make sense.

Comment (vinayak-trivedi) [reply docs/local-backend-roles/localroles.md]: makes sense. I will do that

File: docs/local-backend-roles/localroles.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Setting roles in your local backend
+
+Please make sure that you have setup the backend locally. Watch these videos if you have not done that:
```
Comment (shubham-y) [inline docs/local-backend-roles/localroles.md]:
```suggestion
Please make sure that you have setup the backend locally by running the command `yarn validate-setup`. Watch these videos if you have not done that: 
```

Comment (shubham-y) [reply docs/local-backend-roles/localroles.md]: Or we can add a link to this as well https://github.com/Real-Dev-Squad/website-backend#confirm-correct-configuration-setup

**Review by shubham-y — COMMENTED** [review]
LGTM 👍

**Review by shubham-y — APPROVED** [review]

File: docs/local-backend-roles/localRole.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+- There might be a certain feature that you are building which requires admin or super_user role in order to perform a particular action.
+- Let's take an example: suppose you are working on a feature which allows the super_user to promote a user of Real Dev Squad to member, for that you need to hit `/moveToMembers/:username` endpoint, it uses a authorizeUser middleware to check if the user has a super_user role. For that you need to use your local backend and set up the required roles in firebase to provide super_user role to yourself.
+
+## Steps: 
+
+1. Make sure that your backend server is running. To run the server run `yarn dev` in your project directory. This will run your server on localhost:3000
+
+2. The RDS backend hits `/users/self` endpoint to get the data of the user.
+
+3. So let's try to acces our details by going to http://localhost:3000/users/self. This will give a 401 error, to solve this follow the step below.
```
Comment (rohan09-raj) [inline docs/local-backend-roles/localRole.md:18]: Typo `access`

File: docs/local-backend-roles/localRole.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Setting roles in your local backend
+
+Please make sure that you have setup the backend locally by running the command `yarn validate-setup`. Watch these videos if you have not done that: 
+- <a href="https://www.youtube.com/watch?v=haqPaPRrhPU">Real Dev Squad Backend Setup Part 1: GitHub OAuth and Downloading Firestore credentials json file</a>
+
+- <a href="https://www.youtube.com/watch?v=2Ja5hH2nH1o&feature=youtu.be">Real Dev Squad Backend Setup Part 2: Firestore</a>
+
+## Why we need to step up roles in our local backend?
```
Comment (rohan09-raj) [inline docs/local-backend-roles/localRole.md:8]: Typo `step` -> `setup`

File: docs/local-backend-roles/localRole.md (hunk 4 above)
Comment (rohan09-raj) [reply docs/local-backend-roles/localRole.md:18]: Also can we please have the URLs wrapped with back ticks ?

Comment (rohan09-raj) [reply docs/local-backend-roles/localRole.md:18]: Also, can we please the URL in back tick as you have done for other URLs

File: docs/local-backend-roles/localRole.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+4. Copy this URL `https://github.com/login/oauth/authorize?client_id=<your-client-id>` and paste it somewhere, now go to your  `local.js` (you can see how we got that clientId <a href='https://www.youtube.com/watch?v=haqPaPRrhPU'>here</a>) file and copy the clientId that is present there, and paste it in place of `<your-client-id>` in the URL provided above.
+
+5. This will redirect you to `http://localhost:3000/healthcheck`, if you go to your firebase console, you will notice that a `doc` has been created with your userId in the `users` collection in your firebase, now try going to http://localhost:3000/users/self. you can see those details.
+
+6. You can see that there is no roles property present in the object that is returned. we will update the doc that contains our details with the roles. Click on add fields and set the properties as shown below: 
+
+![Firebase Image](/public/assets/firebase-image.jpg)
+
+7. Now click on add, and now if you go to http://localhost:3000/users/self, you will be able to see the roles.
```
Comment (rohan09-raj) [inline docs/local-backend-roles/localRole.md:28]: Here too, can we wrap this in back ticks ?

Comment (vinayak-trivedi) [reply docs/local-backend-roles/localRole.md:28]: I want it to be clickable so that it redirects them to that link that is why I did not add it

Comment (rohan09-raj) [reply docs/local-backend-roles/localRole.md:28]: Ahh I see, back ticks remove the clicks on a URL

**Pull request merged**